javascript/javascript3/week2/warmup/script.js

```diff
@@ -0,0 +1,116 @@
+//Movies exercise
+//Create an array of bad movies
+const moviesUrl = "https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json"
+const myOl = document.createElement("ol");
+myOl.innerHTML = "Bad Movies";
+document.body.appendChild(myOl);
+
+const latestBadMv = document.createElement("ol");
+latestBadMv.innerHTML = "Bad Movies since 2000";
+document.body.appendChild(latestBadMv);
+fetch(moviesUrl)
+    .then(Response => Response.json())
+    .then(movies => {
+        const badMovies = movies.filter(movie => movie.rating < 4);
+        badMovies.forEach(mv => {
+            const myLi = document.createElement("li");
+            myLi.style.fontSize = "10px";
+            myLi.innerHTML = `${mv.title} with ${mv.rating} rating`;
+            myOl.appendChild(myLi);
+        });
+        //Creat an array of bad movies since year 2000
+        const latestBadMovied = badMovies.filter(movie => movie.year >= 2000);
+        latestBadMovied.forEach(mv => {
+            const latestBadMvLi = document.createElement("li");
+            latestBadMvLi.style.fontSize = "10px";
+            latestBadMvLi.innerHTML = `${mv.title} in year ${mv.year}`;
+            latestBadMv.appendChild(latestBadMvLi);
+        });
+    })
+
+//Promise that resolves after set time & use it with async/await
+async function myDelayMessage(resolveAfter) {
+    await new Promise((resolve, reject) => {
+        if (typeof resolveAfter !== "number") {
+            reject()
+        }
+        setTimeout(function() {
+            resolve()
+        }, resolveAfter * 1000)
+    })
+}
+myDelayMessage(8)
+    .then(() => {
+        console.log("I am called asynchronously"); // logged out after 8 seconds
+    })
+    .catch(() => {
+        console.log("Second must be a number!")
+    });
+
+//Rewrite time
+//Rewrite setTimeout
+function setTimeoutPromise(time) {
+    return new Promise(resolve => {
+        setTimeout(function() {
+            resolve();
+        }, time)
+    })
+}
+setTimeoutPromise(3000).then(() => {
+    console.log("Called after 3 seconds ")
+})
+
+//Rewrite navigator.geolocation.getCurrentPosition
+
+let currentLocation = document.createElement("p")
+document.body.appendChild(currentLocation);
+
+function getCurrentPosition() {
+    return new Promise((resolve, reject) => {
+        if (navigator.geolocation) {
+            resolve();
+        } else {
+            reject();
+        }
+    })
+}
+getCurrentPosition()
+    .then(() => {
+        navigator.geolocation.getCurrentPosition(position => {
+            let latitude = position.coords.latitude;
+            let longitude = position.coords.longitude;
+            currentLocation.innerHTML = `Latitude; ${latitude} 
+           <br>
+           Longitude; ${longitude}`
```

Comment:
this logic, calling the `getCurrentPosition` function, should be inside the Promise at line 69-75 :)